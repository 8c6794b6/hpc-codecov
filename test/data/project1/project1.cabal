cabal-version: 1.12

-- This file has been generated from package.yaml by hpack version 0.33.0.
--
-- see: https://github.com/sol/hpack
--
-- hash: 2a82452cb1b1a322c856e7d346b38bfb8136930e2638507eb703234a0658f2dc

name:           project1
version:        0.1.0.0
description:    Please see the README on GitHub at <https://github.com/githubuser/project1#readme>
homepage:       https://github.com/githubuser/project1#readme
bug-reports:    https://github.com/githubuser/project1/issues
author:         Author name here
maintainer:     example@example.com
copyright:      2021 Author name here
license:        BSD3
license-file:   LICENSE
build-type:     Simple
extra-source-files:
    README.md
    ChangeLog.md

source-repository head
  type: git
  location: https://github.com/githubuser/project1

library
  exposed-modules:
      Lib
  other-modules:
      Paths_project1
  hs-source-dirs:
      src
  build-depends:
      base >=4.7 && <5
  default-language: Haskell2010

executable project1-exe
  main-is: Main.hs
  other-modules:
      Paths_project1
  hs-source-dirs:
      app
  ghc-options: -threaded -rtsopts -with-rtsopts=-N
  build-depends:
      base >=4.7 && <5
    , project1
  default-language: Haskell2010

test-suite project1-test
  type: exitcode-stdio-1.0
  main-is: Spec.hs
  other-modules:
      Paths_project1
  hs-source-dirs:
      test
  ghc-options: -threaded -rtsopts -with-rtsopts=-N
  build-depends:
      base >=4.7 && <5
    , project1
  default-language: Haskell2010
