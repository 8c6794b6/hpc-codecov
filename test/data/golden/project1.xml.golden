<?xml version="1.0" ?><!DOCTYPE coverage SYSTEM 'http://cobertura.sourceforge.nex/xml/coverage-0.4.dtd'><coverage branch-rate="0.25" branches-covered="9" branches-valid="36" complexity="0" line-rate="0.72" lines-covered="18" lines-valid="25" timestamp="0" version="2.0.3"><sources><source>.</source></sources><packages><package name="test.data.project1.src" line-rate="0.875" branch-rate="0.375" complexity="0"><classes><class branch-rate="0.375" complexity="0" filename="test/data/project1/src/Lib.hs" line-rate="0.875" name="Lib"><methods><method name="someFunc" signature="" line-rate="0.0" branch-rate="0.0"><lines><line hits="0" number="13" branch="false"/></lines></method><method name="anotherFunc" signature="" line-rate="0.0" branch-rate="0.0"><lines><line hits="0" number="19" branch="false"/></lines></method></methods><lines><line branch="false" hits="1" number="13"/><line branch="true" hits="1" number="14" condition-coverage="50% (1/2)"/><line branch="false" hits="1" number="15"/><line branch="false" hits="0" number="16"/><line branch="false" hits="1" number="19"/><line branch="true" hits="1" number="20" condition-coverage="50% (3/6)"/><line branch="true" hits="1" number="21" condition-coverage="17% (1/6)"/><line branch="true" hits="1" number="22" condition-coverage="50% (1/2)"/></lines></class></classes></package><package name="test.data.project1.src.Lib" line-rate="0.4444444444444444" branch-rate="0.15" complexity="0"><classes><class branch-rate="0.1" complexity="0" filename="test/data/project1/src/Lib/Bar.hs" line-rate="0.4" name="Lib.Bar"><methods><method name="theNewFunc" signature="" line-rate="0.0" branch-rate="0.0"><lines><line hits="0" number="9" branch="false"/></lines></method></methods><lines><line branch="false" hits="1" number="9"/><line branch="true" hits="1" number="10" condition-coverage="50% (1/2)"/><line branch="true" hits="0" number="11" condition-coverage="0% (0/2)"/><line branch="true" hits="0" number="12" condition-coverage="0% (0/4)"/><line branch="true" hits="0" number="13" condition-coverage="0% (0/2)"/></lines></class><class branch-rate="0.2" complexity="0" filename="test/data/project1/src/Lib/Foo.hs" line-rate="0.5" name="Lib.Foo"><methods><method name="yetAnotherFunc" signature="" line-rate="0.0" branch-rate="0.0"><lines><line hits="0" number="4" branch="false"/></lines></method></methods><lines><line branch="false" hits="1" number="4"/><line branch="true" hits="1" number="5" condition-coverage="50% (2/4)"/><line branch="true" hits="0" number="6" condition-coverage="0% (0/4)"/><line branch="true" hits="0" number="7" condition-coverage="0% (0/2)"/></lines></class></classes></package><package name="test.data.project1.src.Lib.Bar" line-rate="0.5" branch-rate="0.0" complexity="0"><classes><class branch-rate="0.0" complexity="0" filename="test/data/project1/src/Lib/Bar/Buzz.hs" line-rate="0.5" name="Lib.Bar.Buzz"><methods><method name="theReturnedFunc" signature="" line-rate="0.0" branch-rate="0.0"><lines><line hits="0" number="4" branch="false"/></lines></method><method name="&lt;&amp;&amp;&gt;" signature="" line-rate="0.0" branch-rate="0.0"><lines><line hits="0" number="7" branch="false"/></lines></method></methods><lines><line branch="false" hits="1" number="4"/><line branch="false" hits="0" number="7"/></lines></class></classes></package><package name="test.data.project1.test" line-rate="1.0" branch-rate="0.0" complexity="0"><classes><class branch-rate="0.0" complexity="0" filename="test/data/project1/test/Spec.hs" line-rate="1.0" name="Spec"><methods><method name="main" signature="" line-rate="0.0" branch-rate="0.0"><lines><line hits="0" number="4" branch="false"/></lines></method></methods><lines><line branch="false" hits="1" number="4"/><line branch="false" hits="1" number="5"/><line branch="false" hits="1" number="6"/><line branch="false" hits="1" number="7"/><line branch="false" hits="1" number="8"/><line branch="false" hits="1" number="9"/></lines></class></classes></package></packages></coverage>